lmdeploy/pytorch/engine/engine.py

```diff
@@ -636,7 +631,11 @@ async def __long_context_single_forward(inputs):
                 ret['logits'] = ret['logits'][:, last_token_loc]
             return ret
         else:
-            return await __long_context_single_forward(inputs)
+            ret = await __long_context_single_forward(inputs)
+            if not return_logits and not inputs.is_decoding:
+                last_token_loc = [-1]
```

Comment:
Does it influence the following computation?

Reply:
__long_context_single_forward is used for very long prefill. Only last output is required.